internal/controller/supervisorconfig/oidcclientwatcher/oidc_client_watcher.go

```diff
@@ -29,24 +32,28 @@ const (
 	allowedGrantTypesValid = "AllowedGrantTypesValid"
 	allowedScopesValid     = "AllowedScopesValid"
 
-	reasonSuccess              = "Success"
-	reasonMissingRequiredValue = "MissingRequiredValue"
-	reasonNoClientSecretFound  = "NoClientSecretFound"
+	reasonSuccess                  = "Success"
+	reasonMissingRequiredValue     = "MissingRequiredValue"
+	reasonNoClientSecretFound      = "NoClientSecretFound"
+	reasonInvalidClientSecretFound = "InvalidClientSecretFound"
 
 	authorizationCodeGrantTypeName = "authorization_code"
 	refreshTokenGrantTypeName      = "refresh_token"
 	tokenExchangeGrantTypeName     = "urn:ietf:params:oauth:grant-type:token-exchange" //nolint:gosec // this is not a credential
 
-	openidScopeName          = "openid"
-	offlineAccessScopeName   = "offline_access"
+	openidScopeName          = oidc.ScopeOpenID
+	offlineAccessScopeName   = oidc.ScopeOfflineAccess
 	requestAudienceScopeName = "pinniped:request-audience"
 	usernameScopeName        = "username"
 	groupsScopeName          = "groups"
 
 	allowedGrantTypesFieldName = "allowedGrantTypes"
 	allowedScopesFieldName     = "allowedScopes"
 
-	secretTypeToObserve = "storage.pinniped.dev/oidc-client-secret" //nolint:gosec // this is not a credential
+	secretTypeToObserve       = "storage.pinniped.dev/oidc-client-secret" //nolint:gosec // this is not a credential
+	oidcClientPrefixToObserve = "client.oauth.pinniped.dev-"              //nolint:gosec // this is not a credential
+
+	minimumRequiredBcryptCost = 15
```

Comment:
Question: would defining this in terms of the default cost be better?

```suggestion
	minimumRequiredBcryptCost = bcrypt.DefaultCost + 5
```